apis/logging/v1/input_receiver_types.go

```diff
@@ -2,20 +2,37 @@ package v1
 
 // NOTE: The Enum validation on ReceiverSpec.Type must be updated if the list of types changes.
 
+// Receiver type constants, must match JSON tags of OutputTypeSpec fields.
+const (
+	ReceiverTypeHttp   = "http"
+	ReceiverTypeSyslog = "syslog"
+
```

Comment:
I think there should be validation code that rejects mismatched receiver types and receiver type specs, i.e. Type == ReceiverTypeHttp && Syslog != nil.